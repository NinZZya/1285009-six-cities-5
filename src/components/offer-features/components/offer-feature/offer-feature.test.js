import React from 'react';
import renderer from 'react-test-renderer';
import OfferFeature from './offer-feature';

const Feature = {
  ENTIRE: {
    type: `entire`,
    value: `Private room`,
  },
  BEDROOMS: {
    type: `bedrooms`,
    value: `2 Bedrooms`,
  },
  ADULTS: {
    type: `adults`,
    value: `Max 3 adults`,
  },
};

describe(`Should OfferFeature render correctly`, () => {
  it(`Should OfferFeature render correctly type entire`, () => {
    const tree = renderer
      .create(
          <OfferFeature type={Feature.ENTIRE.type} value={Feature.ENTIRE.value} />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });


  it(`Should OfferFeature render correctly type bedrooms`, () => {
    const tree = renderer
      .create(
          <OfferFeature type={Feature.BEDROOMS.type} value={Feature.BEDROOMS.value} />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`Should OfferFeature render correctly type adults`, () => {
    const tree = renderer
      .create(
          <OfferFeature type={Feature.ADULTS.type} value={Feature.ADULTS.value} />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});

