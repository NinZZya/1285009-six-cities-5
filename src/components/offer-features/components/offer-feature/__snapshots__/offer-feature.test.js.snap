// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should OfferFeature render correctly Should OfferFeature render correctly type adults 1`] = `
<li
  className="property__feature property__feature--adults"
>
  Max 3 adults
</li>
`;

exports[`Should OfferFeature render correctly Should OfferFeature render correctly type bedrooms 1`] = `
<li
  className="property__feature property__feature--bedrooms"
>
  2 Bedrooms
</li>
`;

exports[`Should OfferFeature render correctly Should OfferFeature render correctly type entire 1`] = `
<li
  className="property__feature property__feature--entire"
>
  Private room
</li>
`;
